Proposal of database design

kawa18sima opened the pull request:
# WHAT
データベースを設計する。

# WHY
chatspaceを作っていくにあたって、必要な情報をどのようにデータベースで管理するか決めるため。

Commit:
Merge pull request #1 from kawa18sima/Ruby-on-Rails-update

railsのバージョンを「５」に変更

nakashimamasaya committed:
create pull request

nakashimamasaya committed:
change README.md from README.rdoc

To write README with Markdown

nakashimamasaya committed:
README title add

Add title to README

nakashimamasaya committed:
messages table add

In this application, we added the messages table design which is one of
the databases to the README

nakashimamasaya committed:
change messages table in README

Since there were multiple parts of belogns_to, edit
Editing incomplete part of markdown table description

nakashimamasaya committed:
Add description of users table design to README

A description of the user table design of this application was added to
the README.

nakashimamasaya committed:
Add description of groups table design to README

A description of the groups table design of this application was added
to
the README.

nakashimamasaya committed:
Add description of users_groups table design to README

Description of the users_groups table design of this application
README.
Users_groups is an intermediate table.

nakashimamasaya committed:
Change of line break relationship of README

Since the association part was displayed on one line, it changed so as
to be multi-line

nakashimamasaya committed:
Correction of forget symbol of README association description

":" Is given to "through" of the association
To README

kinshotomoya commented on README.md:
```diff
@@ -10,4 +10,16 @@ belongs_to :group
+|column|type|Restriction|
+|:--:|:--:|:--:|
+|name|text|null:false , add_index|
+|email|string|null:false , uniqe:true|
```
emailはdevise導入でデフォルトで入るので書かなくて大丈夫です。

kinshotomoya commented on README.md:
```diff
@@ -10,4 +10,16 @@ belongs_to :group
+|:--:|:--:|:--:|
+|name|text|null:false , add_index|
+|email|string|null:false , uniqe:true|
+|password|integer|null:false , foreign_key:true|
```
passwordも同じ！

kinshotomoya commented on README.md:
```diff
@@ -10,4 +10,16 @@ belongs_to :group
+
+has_many :messages , through :users_groups 
+has_many :groups 
+
```
アソシエーションいい感じ！
あともう一個、中間テーブルともアソシエーション組んでるから書きましょう！

kinshotomoya commented on README.md:
```diff
@@ -22,4 +22,17 @@ has_many :groups
+
+has_many :messages , through :users_groups
+has_many :users
+
```
中間テーブルも書きましょう！

kinshotomoya commented on README.md:
```diff
@@ -22,4 +22,17 @@ has_many :groups
+
+|column|type|Restriction|
+|:--:|:--:|:--:|
+|name|text|null:false , uniqe:true|
```
unique: true　にしなくてもいいんじゃないかな？

kinshotomoya reviewed (commented):
全体的にいい感じ！
二つ伝えたいことがあります！

１　PRの題名は基本的に英語で書いてください！！　エンジニアはみんな英語で書きます！！！！

２　基本的に、同じファイルでの変更は一つのコミットにまとめてください！

頑張りましょおおおおおおおおおおおお

nakashimamasaya committed:
Modification of database design

The changed point is in the README

· The foreign key was changed from integere to references.
· Added association with intermediate table.
· I set the users table to name only.
· Uniqe true of name in group table has been removed.

kawa18sima commented:
ご指摘の点を修正しました。
・中間テーブルとのアソシエーションを追加しました。
・usersテーブルをnameのみにしました。
・groupsテーブルのnameのunique trueを外しました。

あと、
・外部キーの型をintegerからreferencesに変更しました。

よろしくお願いします。

kinshotomoya commented on README.md:
```diff
@@ -33,10 +33,12 @@ has_many :messages , through: :users_groups
+|name|text|null:false|
 
 
 ## users_groups table
```
中間テーブル、user_groups table だね！
前のuserは単数形で！

kinshotomoya reviewed (commented):
コメントしました！
修正いい感じです！
あと二点！！
コメントしたことろを修正したらLGTMで！

修正したら僕に確認取らずにマージして、次進みましょう！！！！！！！！！！！！！！！
次はフロント画面です！
頑張りましょウ！！！！！

nakashimamasaya committed:
Modification of database design

The changed point is in the README

・Changed to users_groups -> user_groups
